src/work-allocation/components/task-assignment/task-assignment.component.ts

```diff
@@ -135,18 +175,24 @@ export class TaskAssignmentComponent implements OnInit {
       });
     } else if (this.location && this.location.id) {
       // Otherwise, get the caseworkers at the specifed location.
-      this.caseworkerService.getForLocation(this.location.id).subscribe(caseworkers => {
-        this.setupCaseworkers(caseworkers);
+      this.caseworkerService.getAll().subscribe(caseworkers => {
+        const locationCaseWorkers = caseworkers.filter(cw => this.locationMatches(cw));
+        this.setupCaseworkers(locationCaseWorkers);
       }, error => {
         handleFatalErrors(error.status, this.router, WILDCARD_SERVICE_DOWN);
       });
     }
   }
 
+  // check if the caseworker's location exists and matches a location
+  public locationMatches(cw: Caseworker): boolean {
```

Comment:
The name of this method doesn't _really_ describe what it's for - if you're checking that a caseworker's location matches the currently selected location, `locationMatches` doesn't really tell the story.

Also, `public` methods (and I'm also not convinced this should be `public` - see other comment as to why) should have `/** */` style comments so that they're properly documented when attempting to use them from another component.

Reply:
Ok, /** */ type comment will be added (and method name changed). Will also make private